Parse and save data

Flix6x opened the pull request.

Flix6x committed:
Move plugin version to top

Flix6x committed:
Slightly expand docstring

Flix6x committed:
Fix typography

Flix6x committed:
Parse response, convert data, save sensors from config and save data from Z-info

Flix6x commented:
I just noticed I forgot to add the logic to disregard beliefs that didn't actually change (these just clutter our database). We're getting a lot of those.

nhoening commented:
Would that logic be best placed more centrally, like FlexMeasures or timely-beliefs?

Flix6x committed:
Fix how sensor data is converted per sensor:

Note: head(-1) doesn't work with a .loc(mask) because you cannot get rid of data that way, and the trailing NaN value remained

Flix6x committed:
Set up BeliefsDataFrame with UTC timezones only

Flix6x committed:
Disregard beliefs that didn't actually change (these just clutter our database)

Flix6x commented:
See https://github.com/SeitaBV/flexmeasures/pull/183 for the util function.

Flix6x committed:
Infer ambiguous datetimes given time-reversed ordered values

nhoening commented on __init__.py:
```diff
@@ -31,22 +42,156 @@
+                f"{zinfo_sensor_name_field} == '{zinfo_sensor_name}'"
+            ).droplevel(zinfo_sensor_name_field)[zinfo_event_value_field]
+
+            # required by timely_beliefs, TODO: check if that still is the case, see https://github.com/SeitaBV/timely-beliefs/issues/64
```
That issue is about initializing a BDF with a Series. Here, it seems you initialize with a DataFrame, so maybe this is not needed.

Flix6x replied:
It was needed for initializing with a DataFrame, too.

nhoening commented on __init__.py:
```diff
@@ -31,22 +42,156 @@
+                )
+                continue
+            sensor = sensor_dict[sensor_name]
+            df_sensor = df.query(
```
Is this duplicated effort? You already built a list of `df_sensor` objects in the previous loop.

nhoening reviewed (approved):
Nice work, only small comments.

I guess it would be a good time to add some helicopter description how Z-Info works to the Readme.
What I find conceptually noteworthy is that we are configuring sensors we expect, and then have to check if those are the sensors we receive back. This is because the Z-Info host has their own configuration which sensors we are allowed to read, and we always (at least in the current state) get all of these back. 
Am I seeing this correctly?

Flix6x committed:
Move readout of config setting to block in which sensor is created

Flix6x committed:
Replace query with loc
